samples/bluetooth/mesh/dfu/distributor/README.rst

```diff
@@ -221,12 +221,12 @@ SMP over Bluetooth authentication
 =================================
 
 By default, the SMP characteristics don't require authentication when using SMP over Bluetooth to access the :ref:`management subsystem <zephyr:mcu_mgr>`.
```

Comment:
This is wrong now, isn't it? Enabling SMP will now indeed require authentication by default? I think this doc should be rewritten to reflect this new default.

Reply:
I haven't changed the default behaviour. The authentication was enabled by default for a very long time (CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN was enabled by default).

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v3.3.0/subsys/mgmt/mcumgr/transport/Kconfig.bluetooth

The CONFIG_MCUMGR_TRANSPORT_BT_AUTHEN option was enabled when the BT_SMP module (used for Bluetooth pairing was enabled). So it is possible that you have authentication disabled if the BT_SMP is disabled in your project

Reply:
Ah right, it looks like I'm a bit confused here. I'm discussing with the team currently

Reply:
This SMP is the Simple Management Protocol (SMP) from MCUmgr: https://docs.zephyrproject.org/latest/services/device_mgmt/mcumgr.html
And `BT_SMP` is the Bluetooth Host Security Manager Protocol.

So I think this section is Ok because `BT_SMP` is still disabled by default: https://github.com/nrfconnect/sdk-zephyr/blob/6f6b0e4ac941540618a5933449538408acc85c7b/subsys/bluetooth/host/Kconfig#L365-L372

What have been changed is that once `BT_SMP` is enabled, the SMP from MCUmgr will require authentication (https://github.com/nrfconnect/sdk-zephyr/pull/2034).

I'm ok with the change. This use-case is covered in CI now so if something changes, mesh CI should fail.